src/services/review/RepoCheckerService.ts

```diff
@@ -0,0 +1,234 @@
+import fs from "fs"
+import path from "path"
+
+import jsdom from "jsdom"
+import { forEach } from "lodash"
+import { marked } from "marked"
+import { ResultAsync, errAsync, ok } from "neverthrow"
+
+import { EFS_VOL_PATH_STAGING } from "@root/constants"
+import SiteCheckerError from "@root/errors/SiteCheckerError"
+import { RepoError } from "@root/types/siteChecker"
+import { ALLOWED_FILE_EXTENSIONS } from "@root/utils/file-upload-utils"
+
+const { JSDOM } = jsdom
+
+export default class RepoCheckerService {
+  getAllRepos(): ResultAsync<string[], Error> {
+    // todo
+    return errAsync(new Error("Not implemented"))
+  }
+
+  runCheckerForAllRepos(): ResultAsync<undefined, Error> {
+    // todo
+    return errAsync(new Error("Not implemented"))
+  }
+
+  async checkRepo(repo: string) {
+    const repoPath = path.join(EFS_VOL_PATH_STAGING, repo)
+
+    const mdFiles = new Set<string>()
+    // get list of all documents
+    const checker = this.getListOfMarkdownFiles(repoPath)
+      .andThen((files) => {
+        files.forEach((file) => {
+          mdFiles.add(file)
+        })
+        console.log("calling getAllDocumentsPath", repoPath)
+        return this.getAllMediaPath(repoPath)
+      })
+      .andThen((setOfAllMediaPath) => {
+        console.log(setOfAllMediaPath)
```

Comment:
This is gonna be noisy if there's alot of paths - are we sure we need this?

Reply:
ps, all console logs rmed